### PR TITLE
Fix broken notifications for mentions from local moderators in 4.2.11

### LINKED ISSUE #31467: /streaming/user and /streaming/user/notification are not working on version 4.2.11
### Steps to reproduce the problem

1. Initiate a Streaming Request:

- Make a GET streaming request to either the /streaming/user/notification or /streaming/user endpoint.

2. Wait for Incoming Data:

-  Keep the connection open and wait for new messages or data to be received through the streaming connection.

3. Observe Received Data:

- Notice that while heartbeat messages appear as expected, mention notifications are not being received.

### Expected behaviour

notifications should appear in stream as expected

### Actual behaviour

no notifications in stream

### Detailed description

I was using "/streaming/user/notification" to manage replays in my elixir based application. However, after last update it stopped to work. Not sure if some params need to changed or if API was changed somehow. I still see heartbeat but no other messages appear.

### Mastodon instance

mastodon.kaiman.uk

### Mastodon version

v4.2.11

### Technical details

I use docker compose with "ghcr.io/mastodon/mastodon:v4.2.11"

Note: not sure if it is related/helpful but I saw this error in logs 

```
sidekiq-1    | 2024-08-16T19:05:52.706Z pid=7 tid=2khf WARN: {"context":"Job raised exception","job":{"retry":true,"queue":"default","class":"LocalNotificationWorker","args":[112933604811709660,17338,"Mention","mention"],"jid":"08bd729d56691160284a8051","created_at":1723834936.1009328,"enqueued_at":1723835152.5992892,"error_message":"undefined method `user_role' for nil:NilClass","error_class":"NoMethodError","failed_at":1723834936.327193,"retry_count":3,"retried_at":1723835034.2850657}}
sidekiq-1    | 2024-08-16T19:05:52.706Z pid=7 tid=2khf WARN: NoMethodError: undefined method `user_role' for nil:NilClass
sidekiq-1    | 2024-08-16T19:05:52.707Z pid=7 tid=2khf WARN: /opt/mastodon/app/services/notify_service.rb:90:in `from_staff?'
sidekiq-1    | /opt/mastodon/app/services/notify_service.rb:116:in `blocked?'
sidekiq-1    | /opt/mastodon/app/services/notify_service.rb:19:in `call'
(indexing_scheduler)

```

### PR DESCRIPTION
**Pull request by ClearlyClaire**
This was caused by an incorrect backport of #31271 to v4.2.11.

### CODE REVIEW

File: app/services/notify_service.rb [hunk 1]
```diff
@@ -87,7 +87,7 @@ def response_to_recipient?
 
   def from_staff?
     sender = @notification.from_account
-    sender.local? && sender.user.present? && sender.user_role&.overrides?(@recipient.user_role) && @sender.user_role&.highlighted? && sender.user_role&.can?(*UserRole::Flags::CATEGORIES[:moderation])
+    sender.local? && sender.user.present? && sender.user_role&.overrides?(@recipient.user_role) && sender.user_role&.highlighted? && sender.user_role&.can?(*UserRole::Flags::CATEGORIES[:moderation].map(&:to_sym))
```
Comment (ThisIsMissEm) [inline app/services/notify_service.rb:90]: Oh wow, so in 4.2 these were strings through `%w` and in 4.3 these are symbols via `%i` and that's what caused this change to break when backported?

Comment (ClearlyClaire) [reply app/services/notify_service.rb:90]: There were two issues, that, and a `@sender` instead of `sender`.

Comment (ThisIsMissEm) [reply app/services/notify_service.rb:90]: oooh, ha, I missed that! Yikes. I do not envy the work of having to backport changes..

Comment (ThisIsMissEm) [reply app/services/notify_service.rb:90]: Actually, why did we decide that the User's role needed to be highlighted to bypass? This kind renders the feature a bit useless for mods who aren't entirely public that they're a moderator.

(We really need #31081 and #27562 which would remove the need for this notification filter bypass feature)

Comment (ClearlyClaire) [reply app/services/notify_service.rb:90]: We did it so that “incognito” mods would not be outed. It was the reason behind the change.

**Review by ThisIsMissEm — APPROVED** [review]
The diff in this PR makes sense to me. I understand that the issue only happened due to backporting from `main`

File: app/services/notify_service.rb (hunk 1 above)
Comment (ThisIsMissEm) [reply app/services/notify_service.rb:90]: Mmm, okay, yeah.. I guess that makes sense.

I guess an operator/admin could always create a separate role of "Moderator" and "Moderator (private)", and make just the "Moderator" one highlighted?

**Review by renchap — APPROVED** [review]

**Review by Goop-arsh — COMMENTED** [review]
Run

**Review by Goop-arsh — COMMENTED** [review]
Run

**Pull request merged**